src/ucs/sys/sock.c

```diff
@@ -824,6 +824,44 @@ ucs_status_t ucs_sock_ipstr_to_sockaddr(const char *ip_str,
     return UCS_ERR_INVALID_ADDR;
 }
 
+static int is_ipv6_str(const char *ip_port_str)
+{
+    /* Address is treated as IPv6 if it contains more than one colon */
+    const char *ch = strchr(ip_port_str, ':');
+    if (NULL != ch) {
+        return (NULL != strchr(++ch, ':'));
+    }
+    return 0;
+}
+
+ucs_status_t ucs_sock_ipportstr_to_sockaddr(const char *ip_port_str,
```

Comment:
can't we use ucs_sock_ipstr_to_sockaddr?

Reply:
BTW, can we rely on `getaddrinfo` to handle both port and IP address?

Reply:
won't it be simpler to just split by ':' and use last argument as port number?
why need the brackets notation with IPv6?

Reply:
AFAIR getaddrinfo can also handle hostnames

Reply:
Yes, it would me much simpler if we could just split by colon.
But there are 2 reasons why we can't do this:
1) IPv6 itself may contain colons (`::1`, `::`)
2) Ok, can we split just the **last** colon and use it as a port?
No, because port argument is optional. It may or may not be present in the config

I'm checking getaddrinfo

Reply:
My understanding is that `getaddrinfo` would not be really helpful here, because it expects to receive a port as a separate argument. But we don't have a port as a separate string, that's why we have parsing - that's the whole thing here to split IP and port from the same string

The complexity of this function comes from the facts that:
1. we use colon as a separator, which interferes with IPv6 string representation
2. IP can be either IPv4 or IPv6
3. port arg is optional

If we really want to make it simpler, then we can use some other symbol as an IP-port separator: (`/`, `;`, `-`, `%`). E.g. `192.168.1.1/13311`
Personally I don't really like it

Reply:
Ok, according to https://en.wikipedia.org/wiki/IPv6_address and https://datatracker.ietf.org/doc/html/rfc3986, we should use square brackets indeed.
Can we add support for getaddrinfo? maybe even in another PR.

Reply:
Yes, let's consider support for `getaddrinfo` in another PR, this one already grew up beyond 500LOC